### ISSUE TITLE
RKE2 - Unable to upgrade rke2 clusters if click on "Config" button

### ISSUE THREAD

**Issue by davidnuzik** [rancher/dashboard#3682]
Rancher version: v2.6-head 878475007 8/5/21
Rancher cluster type: single-node rancher server in docker container
Docker version: 19.03

Downstream cluster type: rke2
Downstream K8s version: v1.21.3-rc2+rke2r2 -> v1.21.3-rc3+rke2r2 upgrade
Note: I had to create my own [custom KDM](https://raw.githubusercontent.com/davidnuzik/kontainer-driver-metadata/qa-temp1/data/data.json) so rc2 was available for testing upgrades. We don't support 1.20 and 1.19 rke2 in Rancher 2.6.0 (at least as of now), so I thought it probably best to use rc2 to start from and rc3 to upgrade to since both work in v2.6-head today. My testing was done with the digital ocean provider.

You can use my KDM link for testing.

**Steps:**
1. Provision an RKE2 cluster using the custom KDM. Pick the rc2 older version. This can be a single node cluster.
2. Once the cluster is ready, from the Cluster Management page, click on the link to navigate to the cluster:
![image](https://user-images.githubusercontent.com/45179589/128405662-eedbc2b6-1d89-4464-a299-1b221b719e3b.png)

3. Then, click on the "Config" button at the top right:
![image](https://user-images.githubusercontent.com/45179589/128405806-1c6b635a-a502-4dbb-97f0-e3e2257f54de.png)

4. _**Now attempt to click on the "Kubernetes Version" dropdown so you can upgrade from rc2 to rc3. You won't be able to click on this drop down. (This is the bug).**_
5. Now, go back to the Cluster Management page, this time instead of navigating to the cluster, click on the function menu (the vertical ...) and click on "Edit Config" from that menu:
![image](https://user-images.githubusercontent.com/45179589/128406010-ad5a16e0-db56-4cbf-a436-677c84a0ebda.png)

This seems to take us to a slightly different version of this edit cluster page.

6. Now, you can access the "Kubernetes Version" drop down and are able to upgrade to rc3.

**Additional information:**
I didn't spot anything useful in rancher server logs or in the web browser dev console.

This issue does not affect RKE1 clusters.

**Comment by davidnuzik** [comment]
This isn't an immediate blocker, just a release blocker IMO for 2.6.0 since there is a workaround as described in the issue.

**Comment by richard-cox** [comment]
This is working as intended, the config page is read only.

**Comment by davidnuzik** [comment]
@deniseschannon why is the Config page read only? That makes no sense to me.

**Comment by deniseschannon** [comment]
@davidnuzik These are the view pages of a resource where you can click on a cluster (the resource) and you can view the resource by either the Detail, Config or Yaml. You aren't editing it cause you didn't select to edit it, but you are just viewing it.

Like other areas in Rancher, you need to actively choose to start editing resources. We've never allowed anyone to immediately start editing without consciously selecting to edit it. 

So this is expected behavior.

**Comment by davidnuzik** [comment]
@deniseschannon That makes sense but the way it looks in the UI is like a carbon copy of the editing screen. We should change the styling to make it more clear this is read only. Maybe mention at the top to go to cluster manager to edit the config (and link). There is a "No" cursor on some parts of some fields/elements but it doesn't work correctly everywhere making it seem like it's editable. The fields are not "grayed out" or anything at least on dark mode. I'll file a separate ticket to request that. I'd argue that sure this is expected but the way it appears is not as expected.

**Comment by davidnuzik** [comment]
Also note RKE1 doesn't have this behavior, if you click the Config button for an rke1 cluster it lets you edit it. This is different behavior for the same button...

**Comment by davidnuzik** [comment]
Closing this issue in favor of https://github.com/rancher/dashboard/issues/3698
